src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1522,7 +1526,7 @@ private BoundBlock BindBlockParts(BlockSyntax node, DiagnosticBag diagnostics)
                 boundStatements.ToImmutableAndFree());
         }
 
-        internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType, BoundExpression expression, DiagnosticBag diagnostics, bool isDefaultParameter = false, RefKind refKind = RefKind.None)
+        internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType, BoundExpression expression, DiagnosticBag diagnostics, bool isDefaultParameter = false, bool isRefVariable = false)
```

Comment:
>isRefVariable [](start = 181, length = 13)

I think this should be called `isRefAssignmetnt` .  The refness of variable is not always interesting because you can have ordinary byval assignments between ref locals.
The important bit here is that the assignment is a `ref` assignment.